src/EditorFeatures/CSharpTest/Diagnostics/FixReturnType/FixReturnTypeTests.cs

```diff
@@ -0,0 +1,276 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.CodeFixes.FixReturnType;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics.FixReturnType
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsFixReturnType)]
+    public partial class FixReturnTypeTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpFixReturnTypeCodeFixProvider());
+
+        [Fact]
+        public async Task Simple()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] 1;
+    }
+}",
+@"class C
+{
+    int M()
+    {
+        return 1;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task Simple_WithTrivia()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    /*A*/ void /*B*/ M()
+    {
+        [|return|] 1;
+    }
+}",
+@"class C
+{
+    /*A*/
+    int /*B*/ M()
+    {
+        return 1;
+    }
+}");
+            // Note: the formatting change is introduced by Formatter.FormatAsync
+        }
+
+        [Fact]
+        public async Task ReturnString()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] """";
+    }
+}",
+@"class C
+{
+    string M()
+    {
+        return """";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnNull()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] null;
+    }
+}",
+@"class C
+{
+    object M()
+    {
+        return null;
+    }
+}");
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/33481")]
+        public async Task ReturnTypelessTuple()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] (null, string.Empty);
+    }
+}",
+@"class C
+{
+    object M()
+    {
+        return (null, string.Empty);
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ReturnLambda()
+        {
+            await TestInRegularAndScript1Async(
+@"class C
+{
+    void M()
+    {
+        [|return|] () => {};
+    }
+}",
+ @"class C
+{
+    object M()
```

Comment:
Is this really helpful? Shouldn't we deduce the actual type here such as `Action`, `Func`, etc. If you feel that is out of scope of the PR, probably worth filing a tracking work item.

Reply:
I agree it's not great. I'll file a follow-up issue.